### PR TITLE
Passing grade

### PR DESCRIPTION
**Pull request by polesye**

### CODE REVIEW

File: cms/djangoapps/contentstore/features/grading.feature [hunk 1]
```diff
@@ -1,134 +0,0 @@
-@shard_1
```
Comment (polesye) [inline cms/djangoapps/contentstore/features/grading.feature]: FYI: converted to bok-choy.

File: cms/djangoapps/contentstore/features/grading.py [hunk 2]
```diff
@@ -1,215 +0,0 @@
-# pylint: disable=missing-docstring
```
Comment (polesye) [inline cms/djangoapps/contentstore/features/grading.py]: FYI: converted to bok-choy.

File: cms/envs/test.py [hunk 3]
```diff
@@ -283,5 +283,4 @@
 # Dummy secret key for dev/test
 SECRET_KEY = '85920908f28904ed733fe576320db18cabd7b6cd'
 
-FEATURES['ENABLE_CUSTOM_GRADING'] = False
```
Comment (polesye) [inline cms/envs/test.py]: FYI: Removed redundand FEATURE flag

File: cms/static/coffee/spec/main.coffee [hunk 4]
```diff
@@ -216,7 +216,7 @@ define([
 
     "coffee/spec/models/course_spec", "coffee/spec/models/metadata_spec",
     "coffee/spec/models/section_spec",
-    "coffee/spec/models/settings_course_grader_spec",
+    "js/spec/models/settings/course_grader_spec",
```
Comment (polesye) [inline cms/static/coffee/spec/main.coffee]: FYI: Converted coffee to js.

File: cms/static/coffee/spec/models/settings_course_grader_spec.coffee [hunk 5]
```diff
@@ -1,39 +0,0 @@
-define ["js/models/settings/course_grader"], (CourseGrader) ->
```
Comment (polesye) [inline cms/static/coffee/spec/models/settings_course_grader_spec.coffee]: FYI: Converted coffee to js.

File: common/lib/xmodule/xmodule/__init__.py [hunk 6]
```diff
@@ -1,14 +0,0 @@
-"""Contains base helper methods used by xmodules"""
```
Comment (polesye) [inline common/lib/xmodule/xmodule/__init__.py]: FYI: Removed redundand code that was added in the 1st PR.

File: common/lib/xmodule/xmodule/course_module.py [hunk 7]
```diff
@@ -1315,48 +1336,7 @@ def grading_context(self):
 
 
         """
-        # If this descriptor has been bound to a student, return the corresponding
```
Comment (polesye) [inline common/lib/xmodule/xmodule/course_module.py]: FYI: moved without changes into `openedx.core.djangoapps.grading_policy.sequential.py`

File: common/lib/xmodule/xmodule/graders.py [hunk 8]
```diff
@@ -115,7 +116,7 @@ def grader_from_conf(conf):
                     del subgraderconf[key]
 
             subgrader = subgrader_class(**subgraderconf)
-            subgraders.append((subgrader, subgrader.category, weight))
+            subgraders.append((subgrader, subgrader.category, weight, passing_grade))
```
Comment (polesye) [inline common/lib/xmodule/xmodule/graders.py]: Each time when we need to pass additional parameters into subgraders, we have to change this class. What if we will pass entire `subgraderconf` config?
@auraz what do you think?

File: lms/djangoapps/courseware/tests/test_grades.py [hunk 9]
```diff
@@ -126,47 +126,6 @@ def _gradesets_and_errors_for(self, course_id, students):
         return students_to_gradesets, students_to_errors
 
 
-class TestMaxScoresCache(ModuleStoreTestCase):
```
Comment (polesye) [inline lms/djangoapps/courseware/tests/test_grades.py]: FYI: moved w/o changes into `openedx.core.djangoapps.grading_policy.tests.test_utils.py`

File: lms/djangoapps/courseware/grades.py [hunk 10]
```diff
@@ -454,214 +175,7 @@ def progress_summary(student, request, course, field_data_cache=None, scores_cli
...
     with manual_transaction():
-        return _progress_summary(student, request, course, field_data_cache, scores_client)
-
-
-# TODO: This method is not very good. It was written in the old course style and
-# then converted over and performance is not good. Once the progress page is redesigned
-# to not have the progress summary this method should be deleted (so it won't be copied).
-@use_custom_grading('progress_summary')
-def _progress_summary(student, request, course, field_data_cache=None, scores_client=None):
-    """
```
Comment (polesye) [inline lms/djangoapps/courseware/grades.py]: FYI: moved w/o changes into `openedx.core.djangoapps.grading_policy.sequential.py`

File: lms/djangoapps/courseware/grades.py [hunk 11]
```diff
@@ -267,186 +168,6 @@ def grade(student, request, course, keep_raw_scores=False, field_data_cache=None
         return grade_summary
 
 
-@use_custom_grading('grade')
-def _grade(student, request, course, keep_raw_scores, field_data_cache, scores_client):
-    """
```
Comment (polesye) [inline lms/djangoapps/courseware/grades.py]: FYI: moved w/o changes into `openedx.core.djangoapps.grading_policy.tests.test_utils.py`

File: lms/djangoapps/courseware/grades.py [hunk 12]
```diff
@@ -3,126 +3,27 @@
...
 from openedx.core.djangoapps.signals.signals import GRADES_UPDATED
-from openedx.core.djangoapps.grading_policy import use_custom_grading
 
 
 log = logging.getLogger("edx.courseware")
 
 
-class MaxScoresCache(object):
-    """
-    A cache for unweighted max scores for problems.
```
Comment (polesye) [inline lms/djangoapps/courseware/grades.py]: FYI: moved w/o changes into `openedx.core.djangoapps.grading_policy.utils.py`

File: lms/djangoapps/courseware/grades.py [hunk 13]
```diff
@@ -454,214 +175,7 @@ def progress_summary(student, request, course, field_data_cache=None, scores_cli
...
-
-def weighted_score(raw_correct, raw_total, weight):
-    """Return a tuple that represents the weighted (correct, total) score."""
-    # If there is no weighting, or weighting can't be applied, return input.
-    if weight is None or raw_total == 0:
-        return (raw_correct, raw_total)
-    return (float(raw_correct) * weight / raw_total, float(weight))
-
-
-def get_score(user, problem_descriptor, module_creator, scores_client, submissions_scores_cache, max_scores_cache):
```
Comment (polesye) [inline lms/djangoapps/courseware/grades.py]: FYI: moved w/o changes into `openedx.core.djangoapps.grading_policy.utils.py`

File: lms/djangoapps/courseware/grades.py [hunk 14]
```diff
@@ -454,214 +175,7 @@ def progress_summary(student, request, course, field_data_cache=None, scores_cli
...
-            'display_name': chapter_module.display_name_with_default,
-            'url_name': chapter_module.url_name,
-            'sections': sections
-        })
-
-    max_scores_cache.push_to_remote()
-    return chapters
-
-
-def weighted_score(raw_correct, raw_total, weight):
```
Comment (polesye) [inline lms/djangoapps/courseware/grades.py]: FYI: moved w/o changes into `openedx.core.djangoapps.grading_policy.utils.py`

File: lms/djangoapps/courseware/views.py [hunk 15]
```diff
@@ -1105,11 +1106,7 @@ def _progress(request, course_key, student_id):
                 })
 
     with grades.manual_transaction():
-        if settings.FEATURES['ENABLE_CUSTOM_GRADING']:
```
Comment (polesye) [inline lms/djangoapps/courseware/views.py]: Remove redundant code that was added in the 1st PR.

File: lms/templates/courseware/progress.html [hunk 16]
```diff
@@ -142,71 +145,8 @@ <h2 class="title">${_("Congratulations, you qualified for a certificate!")}</h2>
             </div>
         </section>
       %endif
-
-      <div class="chapters">
```
Comment (polesye) [inline lms/templates/courseware/progress.html]: FYI: moved w/o changes into `openedx.core.djangoapps.grading_policy.templates.summary.sequential.html`

File: openedx/core/djangoapps/grading_policy/__init__.py [hunk 17]
```diff
@@ -1,42 +0,0 @@
-"""
```
Comment (polesye) [inline openedx/core/djangoapps/grading_policy/__init__.py]: FYI: removed redundant code.

File: common/lib/xmodule/xmodule/graders.py (hunk 8 above)
Comment (auraz) [inline common/lib/xmodule/xmodule/graders.py]: OK for now, refactor in the future

File: common/test/acceptance/tests/lms/test_lms_progress.py [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+
+    def test_passing_info_table_has_correct_values(self):
+        """
+        Scenario: Ensures that the passing information table has correct values
+        Given I have a course with 2 categories ("Homework", "Exam")
+        And "Homework" has passing grade "70" and contains 2 problems
+        And "Exam" has passing grade "0" and contains 1 problem
+        When I pass the 1st Homework's problem (current grade for the category is 50)
+        And I go to the Progress page
+        Then I see that all categories are not possed in the passing information table
```
Comment (auraz) [inline common/test/acceptance/tests/lms/test_lms_progress.py]: Typo: possed.

File: common/test/acceptance/tests/lms/test_lms_progress.py [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+        When I pass the 2nd Homework's problem (current grade for the category is 100)
+        And I go to the Progress page
+        Then I see that all categories are possed in the passing information table
```
Comment (auraz) [inline common/test/acceptance/tests/lms/test_lms_progress.py]: Same here

File: openedx/core/djangoapps/grading_policy/tests/test_instructor_task_integration.py [hunk 20]
```diff
@@ -7,6 +7,7 @@
 import json
 import unittest
 from django.test.utils import override_settings
+from django.conf import settings
```
Comment (auraz) [inline openedx/core/djangoapps/grading_policy/tests/test_instructor_task_integration.py]: Please check if this is needed

**Comment by auraz** [comment]
Minor comments, after that :+1:

**Pull request merged**